src/m365/spo/commands/page/Page.ts

```diff
@@ -11,75 +11,54 @@ export const supportedPageLayouts = ['Article', 'Home', 'SingleWebPartAppPage',
 export const supportedPromoteAs = ['HomePage', 'NewsPage', 'Template'];
 
 export class Page {
-  public static getPage(name: string, webUrl: string, logger: Logger, debug: boolean, verbose: boolean): Promise<ClientSidePage> {
-    return new Promise((resolve: (page: ClientSidePage) => void, reject: (error: any) => void): void => {
-      if (verbose) {
-        logger.logToStderr(`Retrieving information about the page...`);
-      }
-
-      const pageName: string = this.getPageNameWithExtension(name);
-
-      const requestOptions: any = {
-        url: `${webUrl}/_api/web/getfilebyserverrelativeurl('${urlUtil.getServerRelativeSiteUrl(webUrl)}/SitePages/${formatting.encodeQueryParameter(pageName)}')?$expand=ListItemAllFields/ClientSideApplicationId`,
-        headers: {
-          'content-type': 'application/json;charset=utf-8',
-          accept: 'application/json;odata=nometadata'
-        },
-        responseType: 'json'
-      };
-
-      request
-        .get<PageItem>(requestOptions)
-        .then((res: PageItem): void => {
-          if (res.ListItemAllFields.ClientSideApplicationId !== 'b6917cb1-93a0-4b97-a84d-7cf49975d4ec') {
-            reject(`Page ${name} is not a modern page.`);
-            return;
-          }
-
-          try {
-            resolve(ClientSidePage.fromHtml(res.ListItemAllFields.CanvasContent1));
-          }
-          catch (e) {
-            reject(e);
-          }
-        }, (error: any): void => {
-          reject(error);
-        });
-    });
+  public static async getPage(name: string, webUrl: string, logger: Logger, debug: boolean, verbose: boolean): Promise<ClientSidePage> {
+    if (verbose) {
+      logger.logToStderr(`Retrieving information about the page...`);
+    }
+
+    const pageName: string = this.getPageNameWithExtension(name);
+
+    const requestOptions: any = {
+      url: `${webUrl}/_api/web/getfilebyserverrelativeurl('${urlUtil.getServerRelativeSiteUrl(webUrl)}/SitePages/${formatting.encodeQueryParameter(pageName)}')?$expand=ListItemAllFields/ClientSideApplicationId`,
+      headers: {
+        'content-type': 'application/json;charset=utf-8',
+        accept: 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    const res: PageItem = await request.get<PageItem>(requestOptions);
+    if (res.ListItemAllFields.ClientSideApplicationId !== 'b6917cb1-93a0-4b97-a84d-7cf49975d4ec') {
+      throw `Page ${name} is not a modern page.`;
+    }
+
+    return ClientSidePage.fromHtml(res.ListItemAllFields.CanvasContent1);
```

Comment:
the previous implementation handled unexpected errors. I think we should also add a `try.. catch` block with `this.handleRejectedODataJsonPromise(err);`

Reply:
sorry I missed this comment 🙏.
ok your right. I am not sure actually what I had in mind now 😅, maybe I missed that this is actually not part of the command logic 🙂. Let's leave it as it is now 👍